experimenter/experimenter/nimbus-ui/src/components/PageHome/SearchBar/index.test.tsx

```diff
@@ -39,6 +42,35 @@ describe("SearchBar", () => {
       expect(searchInput).toHaveValue("");
     });
   });
+  it("calls resetWindowLocation when nimbus-ui-search is not in localStorage", () => {
+    const resetWindowLocation = jest.fn();
+    // jest.spyOn(Subject.prototype, 'resetWindowLocation').mockImplementation(resetWindowLocation);
+
+    jest.spyOn(window, "localStorage", "get").mockReturnValue({});
+
+    const formRef = { current: {} };
+
+    render(<Subject experiments={[]} onChange={formRef} />);
+
+    expect(resetWindowLocation).toHaveBeenCalledTimes(1);
+  });
+});
+
+describe("resetWindowLocation", () => {
+  it("resets window location to default homepage", () => {
+    // Mock the window.location object
+
+    global.window = Object.create(window);
+    global.window.location = {
+      search: "?query=test",
+    };
+
+    // Call the function
+    resetWindowLocation();
+
+    // Expect window.location to be updated
+    expect(window.location.search).toBe("");
+  });
 });
```

Comment:
try this- 
```javascript
 it("calls resetWindowLocation when nimbus-ui-search is not in localStorage", () => {
  const resetWindowLocation = jest.fn();
  jest.spyOn(Subject.prototype, "resetWindowLocation").mockImplementation(resetWindowLocation);

  jest.spyOn(window.localStorage, "getItem").mockReturnValue(null);

  render(<Subject experiments={[]} onChange={{ current: {} }} />);

  expect(resetWindowLocation).toHaveBeenCalledTimes(1);
});

describe("resetWindowLocation", () => {
  it("resets window location to default homepage", () => {
    // Mock the window.location object
    const oldLocation = window.location;
    delete window.location;
    window.location = { ...oldLocation, search: "?query=test" };

    // Call the function
    resetWindowLocation();

    // Expect window.location to be updated
    expect(window.location.search).toBe("");
  });
});

Reply:
> 

Thank you for the help and suggestion @yashikakhurana .  I tried using the code above and I tried to modify it to work but I still got the following errors

 Cannot spyOn on a primitive value; undefined given coming from `jest.spyOn(Subject.prototype, "resetWindowLocation").mockImplementation(resetWindowLocation); (src/components/PageHome/SearchBar/index.test.tsx:47:14)
` and 

 `TypeError: Illegal invocation` coming from `const url = new URL(`${window.location}`); (src/components/PageHome/SearchBar/index.tsx:41:33)`